polygeist/tools/cgeist/driver.cc

```diff
@@ -360,6 +361,8 @@ static int optimize(mlir::MLIRContext &Ctx,
   CanonicalizerConfig.maxIterations = CanonicalizeIterations;
 
   if (OptLevel != llvm::OptimizationLevel::O0) {
+    PM.addPass(polygeist::createArgumentPromotionPass());
+    
```

Comment:
We will likely need to add or move the LICM pass here, between the argument promotion pass and the detect reduction pass.

Reply:
We can deal with the precise pass ordering in a subsequent PR.